governance/default.go

```diff
@@ -200,6 +211,55 @@ func (gt *GovernanceTallyList) Clear() {
 	gt.items = make([]GovernanceTallyItem, 0)
 }
 
+func (vl *VoteSet) Copy() map[string]VoteStatus {
+	vl.mu.RLock()
+	defer vl.mu.RUnlock()
+
+	ret := make(map[string]VoteStatus)
+	for k, v := range vl.items {
+		ret[k] = v
+	}
+
+	return ret
+}
+
+func (vl *VoteSet) GetValue(key string) VoteStatus {
+	vl.mu.RLock()
+	defer vl.mu.RUnlock()
+
+	return vl.items[key]
+}
+
+func (vl *VoteSet) SetValue(key string, val VoteStatus) {
+	vl.mu.Lock()
+	defer vl.mu.Unlock()
+
+	vl.items[key] = val
+}
+
+func (vl *VoteSet) Import(src map[string]VoteStatus) {
+	vl.mu.Lock()
+	defer vl.mu.Unlock()
+
+	for k, v := range src {
+		vl.items[k] = v
+	}
+}
+
+func (vl *VoteSet) Clear() {
+	vl.mu.Lock()
+	defer vl.mu.Unlock()
+
+	vl.items = make(map[string]VoteStatus)
+}
+
+func (vl *VoteSet) Size() int {
+	vl.mu.RLock()
+	defer vl.mu.RUnlock()
+
+	return len(vl.items)
+}
+
```

Comment:
If above operations are frequently called, you can consider removing `defer` for the performance (though I don't  think so..)

Reply:
@ehnuje Is it because `defer` can cause additional overhead?

Reply:
@andybclee Yes, that can be an issue if this is called frequently.

Reply:
If every operations require locking, how about using `sync.Map`?

Reply:
@ehnuje I'm not sure if this case fits in using `sync.Map`. I read [this article](https://www.reddit.com/r/golang/comments/6vf8bu/syncmap_examples/), but this seems not so good case in using sync.Map.

And all these functions are not called frequently. Only when user voted on something, it can be called. So I think the overhead of using defer is negligible in this case.